src/utils/enum-values.ts

```diff
@@ -10,6 +10,7 @@ export function enumValues<T extends string | number, E extends {}>(enumeration:
       continue;
     }
 
+    // @ts-ignore 4.0.0 claims the second part of the condition is not necessary
```

Comment:
TSC claims it can figure out that this condition is never `true`, but I'm not sure if that's actually the case, because we do this "bidirectional mapping". ![Screen Shot 2020-09-11 at 4 17 26 PM](https://user-images.githubusercontent.com/287060/92982549-7c1d9d80-f453-11ea-8b46-69ffb53a7dcc.png)

Reply:
btw, in line 4 it should maybe be? 

```Typescript
const map = new Map<string, string | number>(Lodash.toPairs(enumeration));
```

🤔

Reply:
This one is touchy my friend, took me quite some time to figure it out :/ The intent is for the enumValues() to guess the type of the enum, string VS number, but I don't that's actually possible.

Not to mention that enums and be mixed with namespaces and such contain function and any other exported const.

The util filters out anything that's not a string or a number in order to let out the namespace's members.

The `enumeration[value as keyof E] === key` check is correct, it checks that the reversed enum (as per the way TS manages enums) contains the key, again in order to not include values from a potential mixed namespace.

I'd say: it works as it is :)